### PR TITLE
WIP: HEOM

### LINKED ISSUE #30: HEOM dynamics module
Ideally, this module would implement both Schrodinger and Heisenberg picture evolution.

Another module should implement the second order time non-local master equation as a special case of HEOM.

### PR DESCRIPTION
**Pull request by shoyer**
Fixes #30

### CODE REVIEW

File: qspectra/dynamics/heom.py [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+    H = np.diag(hamiltonian.E(subspace))
+    liouvillian = -1j * super_commutator_matrix(H)
+
+    # precompute the list of N vectorized projection operators
+    proj_op_left = []
+    proj_op_right = []
+    print 'creating projection operators'
+    for n in xrange(N):
+        proj_op = np.zeros((N, N))
+        proj_op[n, n] = 1
```
Comment (shoyer) [inline qspectra/dynamics/heom.py]: you should get these from the Hamiltonian instead

File: qspectra/dynamics/heom.py [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+        proj_op_left.append(super_left_matrix(proj_op))
```
Comment (shoyer) [inline qspectra/dynamics/heom.py]: let's use a sparse matrix version of these functions

File: qspectra/dynamics/heom.py (hunk 1 above)
Comment (donghyun208) [inline qspectra/dynamics/heom.py]: fixed in https://github.com/shoyer/qspectra/commit/402e1946435dbcd7ecd5a0b8e8eaa84f4f87ba21

File: qspectra/dynamics/heom.py (hunk 2 above)
Comment (donghyun208) [inline qspectra/dynamics/heom.py]: fixed in https://github.com/shoyer/qspectra/commit/6acac1ecc50f2260c068c01906c4354df227476c,
i haven't profiled this yet, it's possible that the matrices that we use to build the tensor are not that sparse to begin with

**Pull request merged**